app/src/main/java/com/example/healthinspector/Fragments/ScanFlow/ScanFragment.java

```diff
@@ -0,0 +1,181 @@
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.StringRequest;
+import com.android.volley.toolbox.Volley;
+import com.budiyev.android.codescanner.CodeScanner;
+import com.budiyev.android.codescanner.CodeScannerView;
+import com.budiyev.android.codescanner.DecodeCallback;
+import com.example.healthinspector.Models.ScannedProduct;
+import com.example.healthinspector.R;
+import com.example.healthinspector.databinding.FragmentScanBinding;
+import com.example.healthinspector.databinding.FragmentUserProfileBinding;
+import com.google.zxing.Result;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.parceler.Parcels;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+
+
+public class ScanFragment extends Fragment {
+
+    private FragmentScanBinding binding;
+    private CodeScanner codeScannerView;
+    private static final String TAG = "ScanFragment";
+    private HashMap<Integer, String> novaGroups = new HashMap<>();
+    private String novaGroup;
+    @Override
+    public void onStart() {
+        super.onStart();
+        novaGroups.put(1,"unprocessed or minimally processed food");
+        novaGroups.put(2,"includes processed culinary ingredient");
+        novaGroups.put(3,"processed food");
+        novaGroups.put(4,"ultra processed food or drink product");
+        novaGroup = "";
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+
+        binding = FragmentScanBinding.inflate(inflater, container, false);
+        View view = binding.getRoot();
+        final Activity activity = getActivity();
+        CodeScannerView scannerView = binding.scannerView;
+        codeScannerView = new CodeScanner(getActivity(), scannerView);
+
+        scannerView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                codeScannerView.startPreview();
+            }
+        });
+        codeScannerView.setDecodeCallback(new DecodeCallback() {
+            @Override
+            public void onDecoded(@NonNull final Result result) {
+                activity.runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        Toast.makeText(getContext(), result.getText(), Toast.LENGTH_SHORT).show();
+                        RequestQueue queue = Volley.newRequestQueue(getContext());
+                        JsonObjectRequest objectRequest = new JsonObjectRequest(Request.Method.GET, getString(R.string.URL) + result.getText(), null, new Response.Listener<JSONObject>() {
```

Comment:
we can move this to a separate function so it's easier to read. Additionally, we can add some top level comments describing how we are parsing and even include a link to sample JSON response that is being parsed